cmd/utils/refactortests/refactortests.go

```diff
@@ -0,0 +1,291 @@
+		if err := refactor(fname); err != nil {
+			fmt.Fprintf(os.Stderr, "Error generating output for %v: %v\n", fname, err)
+			fmt.Printf("Press <return> to continue to next file\n")
+			bufio.NewReader(os.Stdin).ReadString('\n')
+		}
+	}
+}
+
+func refactor(fname string) error {
+	bscript, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return err
+	}
+	script := string(bscript)
+
+	pkg := parser.ParseSource(script)
+	if ast.Check(pkg) > 0 {
+		if err := ast.GetError(pkg); err != nil {
+			return err
+		}
+	}
+
+	var oldScript string
+	var edited bool
+	for script != oldScript {
+		oldScript = script
+		fmt.Printf("Content of the script:\n\n%s\n\n", script)
+		if edited {
+			fmt.Println("The script has been edited, re-running...")
+		}
+		rawResult, err := executeScript(script)
+		if err != nil {
+			return err
+		}
+		if len(rawResult) == 0 {
+			break
+		}
+
+		err = editDataOrQuery(pkg, rawResult)
+		if err != nil {
+			return err
+		}
+
+		edited = true
+		script = ast.Format(pkg)
+	}
+
+	if edited {
+		fmt.Printf("Content of the new script:\n\n%s\n\n", script)
+		fmt.Println("Do you want to update the script (y/n)?")
+		scanner := bufio.NewScanner(os.Stdin)
+		scanner.Scan()
+		if scanner.Err() != nil {
+			return scanner.Err()
+		}
+		ans := scanner.Text()
+		if ans != "y" && ans != "\n" {
+			return nil
+		}
+
+		err = ioutil.WriteFile(fname, []byte(script), 0644)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("%s updated\n", fname)
+	}
+	return nil
+}
+
+func executeScript(script string) (string, error) {
+	querier := querytest.NewQuerier()
+	c := lang.FluxCompiler{
+		Query: script,
+	}
+
+	r, err := querier.C.Query(context.Background(), c)
+	if err != nil {
+		return "", fmt.Errorf("error during compilation, check your script and retry: %v", err)
+	}
+	defer r.Done()
+	results, ok := <-r.Ready()
+	if !ok {
+		return "", fmt.Errorf("error retrieving query result: %s", r.Err())
+	}
+
+	var diffBuf, resultBuf bytes.Buffer
+	var gotErr bool
+	for resultName, result := range results {
+		if resultName == "_test_result" {
+			enc := csv.NewResultEncoder(csv.DefaultEncoderConfig())
+			if _, err := enc.Encode(&resultBuf, result); err != nil {
+				fmt.Fprintln(os.Stderr, errors.Wrap(err, "encoding error while running test script:"))
+			}
+		} else if err := result.Tables().Do(func(tbl flux.Table) error {
+			_, _ = execute.NewFormatter(tbl, nil).WriteTo(&diffBuf)
+			return nil
+		}); err != nil {
+			fmt.Fprintln(os.Stderr, errors.Wrap(err, "error while running test script:"))
+			gotErr = true
+		}
+	}
+
+	if gotErr {
+		fmt.Printf("want/got not equal, printing diff:\n\n")
+		fmt.Printf("%s\n\n", diffBuf.String())
+		return resultBuf.String(), nil
+	}
+
+	fmt.Println("want/got are equal; nothing to do.")
+	return "", nil
+}
+
+func editDataOrQuery(node *ast.Package, rawResult string) error {
+	fmt.Println("Do you want to edit the expected result (1) or the query (2)? (any other to exit)")
```

Comment:
I would think the easiest way to handle this would be to allow the user to update either the query or the input data using an editor of their choice, and then to re-run the script and allow the user to accept whatever output it produces as expected.